dashboard/state_aggregator.py

```diff
@@ -25,6 +26,41 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass(init=True)
+class StateApiResult:
+    # Returned data.
+    data: Union[dict, List[dict], Dict[str, JobInfo]] = None
+    # A list of warnings generated from the API that should be delivered to users.
+    warnings: List[str] = None
+
+
+GCS_QUERY_FAILURE_WARNING = (
+    "Failed to query data from GCS. It is due to "
```

Comment:
... It is due to one of the following:

Reply:
actually. can we differentiate these different errors from grpc reply/exception?

Reply:
Good point. Let me try that out!

Reply:
Addressed  (other than potential node crash (grpc unavailable + timeout), it will just raise an exception )